src/clj/muuntaja/core.clj

```diff
@@ -82,8 +82,9 @@
                 ;; optional guard on empty imput
                 (if (and allow-empty-input-on-decode? decode?)
                   (fn [^InputStream is charset]
-                    (if (pos? (.available is)) ($ is charset)))
-                  $))
+                    (if (and (not (nil? is)) (pos? (.available is))) ($ is charset)))
+                  (fn [^InputStream is charset]
+                    (if-not (nil? is) ($ is charset)))))
```

Comment:
The proposal sounds good. I'd maybe call the option `:allow-empty-request-body?`.